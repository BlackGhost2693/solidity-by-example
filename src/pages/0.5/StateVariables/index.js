import React from "react"
import Example from "../../../components/Example"
import html from "./index.html.js"

export default function(props) {
  return (
    <Example
      title="State Variables"
      description="An example of how to get and set state variables in Solidity"
      html={html}
    />
  )
}
