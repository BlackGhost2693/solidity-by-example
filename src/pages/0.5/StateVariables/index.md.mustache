Variables stored on the blockchain are called `state variables`.

`State variables` are declared inside a contract, but outside functions.

```solidity
{{{StateVariables}}}
```
