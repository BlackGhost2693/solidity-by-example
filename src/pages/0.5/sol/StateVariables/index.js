import React from "react"
import Example from "../../../components/Example"
import html from "./index.html.js"
import { SOL_VERSION } from "../constants"

export default function (props) {
  return (
    <Example
      title="Reading and Writing to a State Variable"
      version={SOL_VERSION}
      description="Reading and Writing to a State Variable"
      html={html}
    />
  )
}
