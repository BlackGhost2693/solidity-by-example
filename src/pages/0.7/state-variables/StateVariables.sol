// SPDX-License-Identifier: MIT
pragma solidity ^0.7.6;

contract SimpleStorage {
    // State variable to store a number
    uint public num;

    // You need to send a transaction to write to a state variable.
    function set(uint _num) public {
        num = _num;
    }

    // You can read from a state variable without sending a transaction.
    // Actually we don't need this function. The compiler automatically
    // creates getter functions for all public variables.
    function get() public view returns (uint) {
        return num;
    }
}
