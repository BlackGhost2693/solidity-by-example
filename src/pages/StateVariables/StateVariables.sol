pragma solidity ^0.5.16;

contract SimpleStorage {
    // State variable to store a number
    uint public num;

    // You need to send a transaction to write to a state variable.
    function set(uint _num) public {
        num = _num;
    }

    // You can read from a state varaible without sending a transaction.
    // Actually we don't need this function. The compiler automatically
    // creates getter functions for all public variables.
    function get() public view returns (uint) {
        return num;
    }
}
