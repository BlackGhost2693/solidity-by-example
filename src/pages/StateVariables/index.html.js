const html = `<p>You need to send a transaction to the blockchain to write or update a state variable.</p>
<p>This means that you will have to pay a transaction fee.</p>
<p>On the other hand, you can read a data from a state variable, for free, without sending any transaction.</p>
<pre><code class="language-solidity">pragma solidity ^0.5.16;

contract SimpleStorage {
    // State variable to store a number
    uint public num;

    // You need to send a transaction to write to a state variable.
    function set(uint _num) public {
        num = _num;
    }

    // You can read from a state varaible without sending a transaction.
    // Actually we don&#39;t need this function. The compiler automatically
    // creates getter functions for all public variables.
    function get() public view returns (uint) {
        return num;
    }
}
</code></pre>
`

export default html
