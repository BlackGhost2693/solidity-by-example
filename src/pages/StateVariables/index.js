import React, { Component } from "react"
import Example from "../../components/Example"
import html from "./index.html.js"

class StateVariables extends Component {
  render() {
    return <Example title="State Variables" html={html} />
  }
}

export default StateVariables
