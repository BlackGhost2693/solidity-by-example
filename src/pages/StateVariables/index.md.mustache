Variables declared inside the contract but outside functions are stored on the blockchain.

```solidity
{{{StateVariables}}}
```
