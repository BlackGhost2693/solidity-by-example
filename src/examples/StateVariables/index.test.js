import React from "react"
import { shallow } from "enzyme"
import StateVariables from "./index"

test("it renders", () => {
  const component = shallow(<StateVariables />)

  expect(component).toMatchSnapshot()
})
