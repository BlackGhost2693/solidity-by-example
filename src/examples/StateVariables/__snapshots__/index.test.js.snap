// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<Example
  html="<p>State variables are variables which are permanently stored on the blockchain.</p>
<pre><code class=\\"language-solidity\\">pragma solidity ^0.5.3;

contract SimpleStorage {
    // Variables declared here, inside the contract but outside the
    // functions will be stored on the blockchain.
    string public text;

    // Function to get text.
    function get() public view returns (string memory) {
        return text;
    }
    // Actually we don&#39;t need the function above. The compiler automatically
    // creates getter functions for all public state variables.
    // The compiler will create a function called text().

    // Function to set text.
    function set(string memory _text) public {
        text = _text;
    }
}
</code></pre>
"
  title="State Variables"
/>
`;
